app/Controllers/Http/LandingController.ts

```diff
@@ -30,12 +30,7 @@ export default class LandingController {
     if (domainBeingAccessed === this.mainHostingDomain || domainBeingAccessed === 'localhost') {
       // if domain set using path, use that
       if (typeof params.domainAsPath === 'string') {
-        if (request.url().endsWith('/privacy-policy')) {
-          return response
-            .redirect()
-            .status(301)
-            .toPath('https://' + this.mainHostingDomain + '/privacy-policy')
-        }
+        this.checkRouteForRedirect(request, response)
```

Comment:
@johnnycho I think you need to actually `return` there, otherwise the script wont break/stop on that `if` statement, shouldn't you?

If that's the case, you may want to move this into it's own method in the crontroller and use it in the router file instead the `index` one

Reply:
@TonnyORG The `checkRouteForRedirect()` method contains a `return` with a `response.redirect()` so the script will definitely break on that line if the right conditions are met.

The logic needs to happen in the `index()` method in the controller because that's where a lot of the routing logic currently exists, e.g. for handling `eth.xyz/name.eth` and `eth.xyz/subdomain.name.eth`. Of course I don 't know if it's possible to move that routing logic out of the controller into a separate method or into some kind of middleware, but that's a bigger refactor than I was comfortable with for this task.

Reply:
@johnnycho I'm talking about this line `Route.get('/:domainAsPath?/privacy-policy', 'LandingController.index')` in the `start/routes.ts` file.

Instead of just pointing it to `LandingController.index` point it to `LandingController.redirectPrivacyPolicy` and then that method would handle the redirect itself instead of having the same logic checking for `request.url().endsWith('/privacy-policy'` all around.

I think the logic you're talking about for the domain vs path is already handled at this point by the router itself isn't it? am I missing something?

Reply:
@TonnyORG Excellent suggestion, thank you. I had to make a few other tweaks but it works and is much cleaner.

And now if we want to add other such routes, like `terms-of-use`, we can simply route to that same method and create an Edge template called `terms_of_use.edge`.